aws/resource_aws_servicecatalog_product.go

```diff
@@ -0,0 +1,450 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/servicecatalog"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func resourceAwsServiceCatalogProduct() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsServiceCatalogProductCreate,
+		Read:   resourceAwsServiceCatalogProductRead,
+		Update: resourceAwsServiceCatalogProductUpdate,
+		Delete: resourceAwsServiceCatalogProductDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(15 * time.Minute),
+			Update: schema.DefaultTimeout(15 * time.Minute),
+			Delete: schema.DefaultTimeout(15 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"distributor": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"owner": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"product_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"has_default_path": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"product_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"support_description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"support_email": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"support_url": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"provisioning_artifact": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"description": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"info": {
+							Type:     schema.TypeMap,
+							Required: true,
+							Elem:     schema.TypeString,
+							ForceNew: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+							Default:  servicecatalog.ProvisioningArtifactTypeCloudFormationTemplate,
+						}, // CLOUD_FORMATION_TEMPLATE  | MARKETPLACE_AMI | MARKETPLACE_CAR
```

Comment:
add validation that type is onf the following commented types:

```
ValidateFunc: validation.StringInSlice([]string{
servicecatalog.ProvisioningArtifactTypeCloudFormationTemplate,
servicecatalog.ProvisioningArtifactTypeMarketplaceAmi,
servicecatalog.ProvisioningArtifactTypeMarketplaceCar,
},false)
```

Reply:
:+1: - very helpful TY